UI/window-basic-main.hpp

```diff
@@ -199,6 +200,7 @@ class OBSBasic : public OBSMainWindow {
 	friend struct BasicOutputHandler;
 	friend struct OBSStudioAPI;
 	friend class ScreenshotObj;
+	friend class OBSSystemTrayIcon;
```

Comment:
This should be avoided.


I didn't check yet what forces to add that but I'm sure it can be avoided, `friend` is a last resort.

Reply:
The triggered slots for the outputs were private in OBSBasic.

Reply:
For that add a signal in `OBSSystemTrayIcon` (you can check `OBSBasicControls`) and connect this signal and the private slot in OBSBasic.

Reply:
I've added signals to the system tray. Now the only reason it still needs to be a friend class is the open projector functions.

Reply:
So `AddProjectorMenuMonitors` is the issue.
I don't know if re-creating `previewProjector` and `studioProgramProjector` is necessary, I need to understand more why.

But I think those QMenu mainly containing things from OBSBasic, could be created in OBSBasic and given to the `OBSSystemTrayIcon` instance when needed (e.g. `trayIcon->setPreviewProjectorMenu(createdMenu)` and `trayIcon->setStudioProgramProjectorMenu(createdMenu2)` called from OBSBasic).